### PR TITLE
no more clientp* data for callbacks

### PR DESCRIPTION
**Pull request by wolfv**

### CODE REVIEW

File: src/target.cpp [hunk 1]
```diff
@@ -78,22 +78,22 @@ namespace powerloader
...
 
-    CbReturnCode Target::call_endcallback(TransferStatus status)
+    CbReturnCode Target::call_end_callback(TransferStatus status)
     {
         reset_file(status);
 
-        end_callback end_cb = override_endcb ? override_endcb : target->endcb;
-        void* cb_data = override_endcb ? override_endcb_data : target->cbdata;
+        end_callback_t end_cb
+            = override_end_callback ? override_end_callback : target->end_callback;
```
Comment (Klaim) [inline src/target.cpp:86]: Potential minor improvement: if either `override_end_callback` or `target->end_callback` are not needed anymore after being called, moving them would be potentially less expensive.

**Review by Klaim — APPROVED** [review]

File: src/target.cpp (hunk 1 above)
Comment (wolfv) [reply src/target.cpp:86]: hmm, not sure if moving works, but we could take a reference or a pointer?

Comment (Klaim) [reply src/target.cpp:86]: A reference would work fine here yes.

Comment (wolfv) [reply src/target.cpp:86]: I made the left hand side a reference.

**Pull request merged**